Sources/SwiftProtobuf/TextFormatScanner.swift

```diff
@@ -1090,9 +1095,26 @@ internal struct TextFormatScanner {
             }
             throw TextFormatDecodingError.unknownField
         case asciiLowerA...asciiLowerZ,
-             asciiUpperA...asciiUpperZ,
-             asciiOne...asciiNine: // a...z, A...Z, 1...9
+             asciiUpperA...asciiUpperZ: // a...z, A...Z
             return parseIdentifier()
+        case asciiOne...asciiNine:  // 1...9 (field numbers are 123, not 0123)
+            let start = p
+            p += 1
+            while p != end {
+                let c = p[0]
+                if c < asciiZero || c > asciiNine {
+                    break
+                }
+                p += 1
+                if p - start > maxFieldNumLength {
```

Comment:
This one could also do the calculation if desired and check the specific values instead of being length based.  But most map<> caes will use the key names, the performance impact should be small.

Reply:
I don't need to do that.  I just wanted to be sure we didn't leave an opening for someone to force an integer overflow in this code.